android/src/test/java/com/magicleap/magicscript/scene/nodes/dropdown/UiDropdownListNodeTest.kt

```diff
@@ -0,0 +1,161 @@
+                return providerTypeface
+            }
+        }
+        iconsRepo = mock()
+        val sampleIcon = context.getDrawable(R.drawable.add)
+        whenever(iconsRepo.getIcon(anyString(), anyBoolean())).thenReturn(sampleIcon)
+
+        item1 = buildDropdownItem(reactMapOf().id("0").label("item 1"))
+        item2 = buildDropdownItem(reactMapOf().id("1").label("item 2"))
+
+        tested = UiDropdownListNode(JavaOnlyMap(), context, mock(), fontProvider, iconsRepo)
+        tested.build()
+        tested.addContent(item1)
+        tested.addContent(item2)
+    }
+
+
+    @Test
+    fun `should notify listener about all selected items when multi select active`() {
+        tested.update(reactMapOf(UiDropdownListNode.PROP_MULTI_SELECT, true))
+        val item3 = buildDropdownItem(
+            reactMapOf().id("312").label("item 3").selected(true)
+        )
+        tested.addContent(item3)
+        var selectedItemsList = listOf<UiDropdownListItemNode>()
+        tested.onSelectionChangedListener = { selectedItems ->
+            selectedItemsList = selectedItems
+        }
+
+        item2.update(reactMapOf().selected(true))
+
+        selectedItemsList.size shouldEqual 2
+        selectedItemsList shouldContain item2
+        selectedItemsList shouldContain item3
+    }
+
+    @Test
+    fun `should select only the last clicked item when multi select disabled`() {
+        tested.update(reactMapOf(UiDropdownListNode.PROP_MULTI_SELECT, false))
+
+        item1.update(reactMapOf().selected(true))
+        item2.update(reactMapOf().selected(true))
+
+        item1.isSelected shouldBe false
+        item2.isSelected shouldBe true
+    }
+
+    @Test
+    fun `should be able to select multiple items when multi select enabled`() {
+        tested.update(reactMapOf(UiDropdownListNode.PROP_MULTI_SELECT, true))
+
+        item1.update(reactMapOf().selected(true))
+        item2.update(reactMapOf().selected(true))
+
+        item1.isSelected shouldBe true
+        item2.isSelected shouldBe true
+    }
+
+    @Test
+    fun `should hide dropdown list when item selected if multi select option disabled`() {
+        val props = reactMapOf(
+            UiDropdownListNode.PROP_MULTI_SELECT, false,
+            UiDropdownListNode.PROP_SHOW_LIST, true
+        )
+        tested.update(props)
+
+        item1.update(reactMapOf().selected(true))
+
+        listNode shouldNotBe null
+        listNode?.isVisible shouldBe false
+    }
+
+    @Test
+    fun `should not hide dropdown list when item selected if multi select enabled`() {
+        val props = reactMapOf(
+            UiDropdownListNode.PROP_MULTI_SELECT, true,
+            UiDropdownListNode.PROP_SHOW_LIST, true
+        )
+        tested.update(props)
+
+        item1.update(reactMapOf().selected(true))
+
+        listNode shouldNotBe null
+        listNode?.isVisible shouldBe true
+    }
+
+    private fun buildDropdownItem(props: JavaOnlyMap): UiDropdownListItemNode {
```

Comment:
it may look better in class if as a parameter you'll take an `index` and create props inside:
`reactMapOf().id(index).label("item $index").selected(true)` and in tests: `buildDropdownItem("3")`
what do you think?